exercises/practice/bob/.approaches/method-based-if-statements/content.md

```diff
@@ -0,0 +1,96 @@
+# Method-Based `if` statements
+
+```java
+class Bob {
+    String hey(String input) {
+        var inputTrimmed = input.trim();
+        
+        if (isSilent(inputTrimmed)) {
+           return "Fine. Be that way!";
+        }
+        if (isShouting(inputTrimmed) && isQuestioning(inputTrimmed)) {
+            return "Calm down, I know what I'm doing!";
+        }
+        if (isShouting(inputTrimmed)) {
+            return "Whoa, chill out!";
+        }
+        if (isQuestioning(inputTrimmed)) {
+            return "Sure.";
+        }
+            
+        return "Whatever.";
+    }
+
+    private boolean isShouting(String input) {
+        return input.chars()
+                    .anyMatch(Character::isLetter) &&
+                input.chars()
+                    .filter(Character::isLetter)
+                    .allMatch(Character::isUpperCase);
+    }
+
+    private boolean isQuestioning(String input) {
+        return input.endsWith("?");
+    }
+
+    private boolean isSilent(String input) {
+        return input.length() == 0;
+    }
+}
+```
+
+In this approach, the different conditions for Bob’s responses are separated into dedicated private methods within the `Bob` class. This method-based approach improves readability and modularity by organizing each condition check into its own method, making the main response method easier to understand and maintain.
```

Comment:
Could you please format this paragraph so that each sentence starts on a new line?